src/expr/src/expr/mod.rs

```diff
@@ -148,6 +150,15 @@ pub fn build_from_prost(prost: &ExprNode) -> Result<BoxedExpression> {
             ArrayConcatExpression::try_from(prost).map(Expression::boxed)
         }
         Vnode => VnodeExpression::try_from(prost).map(Expression::boxed),
+        Now => {
```

Comment:
A bit afraid this change will affect use of now in streaming. 

Currently, I think the frontend will Err if the now appears in an invalid location. However, this change may cause the now in stream to evaluate to the static session_timestamp value.

Reply:
Actually planner tests don't help here as the planner will still evaluate it to `Now` expr.

Reply:
> A bit afraid this change will affect use of now in streaming.
> 
> Currently, I think the frontend will Err if the now appears in an invalid location. However, this change may cause the now in stream to evaluate to the static `session_timestamp` value. I don't think this is intended.

I have just added some check and a planner test for it.